iotedgehubdev/certutils.py

```diff
@@ -498,3 +484,85 @@ def _get_kwargs_string(self, kwarg_key, min_length, max_length, default_str=None
                                                          min_length, max_length)
                 raise EdgeValueError(msg)
         return result_str
+
+    def _dump_cert_content(self, id_str, output_path, overwrite_existing=True):
+        if not overwrite_existing and os.path.exists(output_path):
+            raise EdgeFileAccessError(
+                'Following cert file already exists.', output_path)
+        Utils.mkdir_if_needed(os.path.dirname(output_path))
+        cert_dict = self._get_cert_dict(id_str)
+        cert_obj = cert_dict['cert']
+        try:
+            with open(output_path, 'w') as output_file:
+                output_file.write(crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                                          cert_obj).decode('utf-8'))
+        except IOError as ex:
+            msg = 'IO Error when exporting certs for ID: {0}.\n' \
+                  ' Error seen when exporting file {1}.' \
+                  ' Errno: {2} Error: {3}'.format(id_str, ex.filename, str(ex.errno), ex.strerror)
+            raise EdgeFileAccessError(msg, output_path)
+
+    def _dump_cert_key(self, id_str, output_path, overwrite_existing=True):
```

Comment:
do we need overwrite_existing?

Reply:
how about passing cert_dict instead of id_str?

Reply:
I leave this option in case there's requirement to fail the function if file exist in the future. Similar to the 'x' mode in file io library. By default it will overwrite existing files and people can ignore this parameter.

Reply:
is it a valid scenario in our case?

Reply:
besides, currently it is not used, and if needed later, the code still could be added a default value with no backward impact